File: python/spartacus_search.py
def spartacus_search(haystack, needle):
    results = []
    for x in haystack:
        print(x)
        results.append(x)
    return results


if __name__ == "__main__":
    print(spartacus_search([1, 2, 3, 4, 5], 3))
    print(spartacus_search(["a", "b", "c"], "z"))

File: lua/spartacus_search.lua
local function spartacus_search(haystack, _needle)
    local results = {}
    for i, x in ipairs(haystack) do
        print(x)
        results[i] = x
    end
    return results
end

local results = spartacus_search({1, 2, 3, 4, 5}, 3)
print("found " .. #results .. " matches")

return spartacus_search
